src/fragmentation/fragment.rs

```diff
@@ -101,8 +105,9 @@ impl Fragment {
         &mut self,
         fragmentation_property: &String,
         server_address: &String,
+        ldes_id: &str,
     ) -> (Fragment, Fragment) {
-        self.materialize().await;
+        self.materialize(ldes_id).await;
```

Comment:
Since it will always be the same maybe it should be a property of the fragment. But not sure.